### ISSUE TITLE
Request: support to ignore files/dirs from .gitignore 

### ISSUE THREAD

**Issue by jodaka** [svent/sift#13]
I'd love sift to understand .gitignore or similar files for other VCS.
Doc suggest using `sift --exclude-dirs '.git' --write-config` so I could probably manually transfer contents of .gitignore to the sift config, but the problem is: .gitconfig is different for every project while sift config seems to be the same for all projects.

**Comment by svent** [comment]
Supporting .gitignore files is definitely on my todo list. I would like to support the complete pattern options of gitignore files (which are quite complex), so this may take some time but it will get implemented in a future release.

**Comment by netei** [comment]
+1 on this, this is probably the number 1 thing that makes me stay on ag

**Comment by carlsverre** [comment]
+1 also the reason I will likely stay on ag

**Comment by sherwinyu** [comment]
:+1: for this as well! This would make it a great drop in replacement -- right now, this is 3-4x slower than `ag` for one of my projects because of all the extraneous directories being searched

**Comment by ldong** [comment]
Exactly, I would like to see `sift` as a drop-in replacement for `ag` with faster processing speed.

**Comment by adouzzy** [comment]
+1 can't wait for this function.

**Comment by svent** [comment]
Thanks for the support on this, it does help prioritize feature development.

To give an update: I finished the development of the core functionality for this and will now integrate it into sift - I hope to release a version with gitignore support in the near future.

**Comment by svent** [comment]
I just released version 0.6.0 which includes support for .gitignore files via the option `--git`. The overall processing speed is not always faster than not using the new option, as determining whether to search a file or not requires some additional CPU time. sift will now recurse into directories in parallel to compensate for this. The implementation should perform quite well, a future version might include some additional improvements.

As always, somethhing like `sift --git --write-config` can be used to enable this by default.

The implementation took some time as I wanted to support the full .gitignore pattern syntax described [here](https://git-scm.com/docs/gitignore). It supports
- multiple .gitignore files (in parent directories)
- combinations of exclude and include ("!...") patterns with correct prioritization
- independence of the working directory, i.e. it does not matter whether you search a project path while it is your current directory or from somewhere else, the result will be the same

I created a test repository containing various test cases to test tools (and especially sift) supporting .gitignore files: https://github.com/svent/gitignore-test
